### PR TITLE
feat: add Python 3.13 support

### PR DESCRIPTION
**Pull request by jkowalleck**

### CODE REVIEW

**Review by madpah — APPROVED** [review]

**Comment by jkowalleck** [comment]
some code branches are breaking in py312, others display deprecation warnings. 
will investigate and fix, so that we might release a version that is completely supporting py312.

File: tests/_data/snapshots/environment/plain_normalize-packagename_1.0.xml.bin [hunk 1]
```diff
@@ -8,13 +8,6 @@
       <purl>pkg:pypi/ruamel.yaml@0.18.5</purl>
       <modified>false</modified>
     </component>
-    <component type="library">
```
Comment (jkowalleck) [inline tests/_data/snapshots/environment/plain_normalize-packagename_1.0.xml.bin:11]: unexpected to be gone. need to investigate

**Comment by jkowalleck** [comment]
py3.13 is working in general. no issues there. 
but the tests are breaking and appear inconsistent. 

py3.13 or some related thing changed how XML is done ...
```text
-       <description>A small Python package for determining appropriate platform-specific dirs, e.g. a &quot;user data dir&quot;.</description>
?                                                                                                      ^^^^^^             ^^^^^^
+       <description>A small Python package for determining appropriate platform-specific dirs, e.g. a "user data dir".</description>
?                                                                                                      ^             ^
```

possible solution: modify the LIB, so taht every user input is in a CDATA block
i dont like this solution ... meee ... need to think about it.

**Comment by jkowalleck** [comment]
> py3.13 is working in general. no issues there. but the tests are breaking and appear inconsistent.
> 
> py3.13 or some related thing changed how XML is done ...
> 
> ```
> -       <description>A small Python package for determining appropriate platform-specific dirs, e.g. a &quot;user data dir&quot;.</description>
> ?                                                                                                      ^^^^^^             ^^^^^^
> +       <description>A small Python package for determining appropriate platform-specific dirs, e.g. a "user data dir".</description>
> ?                                                                                                      ^             ^
> ```
> 
> possible solution: modify the LIB, so taht every user input is in a CDATA block i dont like this solution ... meee ... need to think about it.

found a solution: 
https://github.com/CycloneDX/cyclonedx-python/blob/d5b68fed7f5608c18f4043301aa34e6281475a92/tests/__init__.py#L152-L159

**Pull request merged**